apps/browser-extension-wallet/src/hooks/useWalletManager.ts

```diff
@@ -752,14 +870,44 @@ export const useWalletManager = (): UseWalletManager => {
     [cardanoWallet]
   );
 
+  const generateSharedWalletKey = useCallback(
+    async (password: string): Promise<Wallet.Crypto.Bip32PublicKeyHex> => {
+      const chainId = getCurrentChainId();
+      const mnemonic = await getMnemonic(Buffer.from(password));
+      const keyAgent = await Wallet.KeyManagement.InMemoryKeyAgent.fromBip39MnemonicWords(
+        {
+          chainId,
+          getPassphrase: async () => Buffer.from('password', 'utf8'),
```

Comment:
it's a good practice to overwrite the secret in buffer once it's no longer needed, so at the end of this function. Same in the `createWallet` function. 
```suggestion
const buffer = Buffer.from(password, 'utf8');
...
          getPassphrase: async () => buffer,
...it's a good practice to overwrite the secret in buffer once it's no longer needed, so at the end of this function
buffer.fill(0);
```

Reply:
Fixed on aaf64f5